specification/types.md

```diff
@@ -128,12 +129,15 @@ This structure is populated by a provider for use by an [Application Author](./g
 
 An enumeration of possible provider states.
 
-| Status    | Explanation                                                                                         |
-| --------- | --------------------------------------------------------------------------------------------------- |
-| NOT_READY | The provider has not been initialized.                                                              |
-| READY     | The provider has been initialized, and is able to reliably resolve flag values.                     |
-| ERROR     | The provider is initialized but is not able to reliably resolve flag values.                        |
-| STALE     | The provider's cached state is no longer valid and may not be up-to-date with the source of truth.  |
+| Status           | Explanation                                                                                        |
+| ---------------- | -------------------------------------------------------------------------------------------------- |
+| NOT_READY        | The provider has not been initialized.                                                             |
+| READY            | The provider has been initialized, and is able to reliably resolve flag values.                    |
+| ERROR            | The provider is initialized but is not able to reliably resolve flag values.                       |
+| STALE            | The provider's cached state is no longer valid and may not be up-to-date with the source of truth. |
+| CONTEXT_PENDING* | The provider is reconciling its state with a context change.                                       |
```

Comment:
Should `CONTEXT_PENDING` also be an error code for Evaluation API? Once a new context is passed into `setEvaluationContext()` I'd expect the Evaluation APIs to not return values associated to the older context and rather force "default" until reconciliation is complete

Reply:
I wouldn't consider this an error in all cases. I think I would defer to the provider on this. They could throw, or return the previous value with a custom reason (we could even add `CONTEXT_PENDING` (or whatever language we end up with) as a pre-defined reason).

Interested in more thoughts on this.

Reply:
I agree on deferring to the Provider on this.
In the cases where the Provider wants to return the default value during reconciliation, we could suggest to use `ERROR` reason with `PROVIDER_NOT_READY` error code (this is how we handle it now in our [Kotlin Confidence Provider](https://github.com/spotify/confidence-openfeature-provider-kotlin/pull/123/files), for example). In this case, we can expand the `PROVIDER_NOT_READY` error code's description:
```
The provider has not been initialized, or a context reconciliation is in progress.
```
Perhaps better to discuss this minor detail on separate PR, and happy to resolve this comment thread.